EVG-18225: Check status code and report error while getting signed ur…

mridul-augustine opened the pull request:
…l in poplar

julianedwards commented on rpc/client.go:
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return "", errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
```
Can you change this to `"failed to get the Data-Pipes signed URL with response '%s'", response.Status`.

julianedwards commented on rpc/client.go:
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
 
 	defer response.Body.Close()
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
```
I would add more context to this message.

julianedwards commented on rpc/client.go:
```diff
@@ -225,6 +232,14 @@ func uploadTestReport(signedURL string, data []byte, client *http.Client) error
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
```
Can you change this to `"failed to upload test report to the Data-Pipes service with response '%s'", response.Status`.

julianedwards commented on rpc/client.go:
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
+
 	body, error := ioutil.ReadAll(response.Body)
 	if error != nil {
 		return "", err
```
I would wrap this error.

julianedwards replied:
Also, on L214 we aren't properly checking the error, you can change it to:
```
if err = json.Unmarshal(body, &responseBody); err != nil {
        return "", errors.Wrap(err, "parsing signed URL response")
}
```

julianedwards commented on rpc/client.go:
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
+		return "", errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
+	}
+
 	body, error := ioutil.ReadAll(response.Body)
```
`err`

julianedwards commented on rpc/client.go:
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
+	}
+
 	body, error := ioutil.ReadAll(response.Body)
 	if error != nil {
```
`if err`

julianedwards reviewed (changes requested):
I think all of the debug logs are a bit overkill, but if you find them useful feel free to keep them.

mridul-augustine commented:
> I think all of the debug logs are a bit overkill, but if you find them useful feel free to keep them.

I just added one debug :) others were just the old ones. Just added more context

julianedwards reviewed (changes requested):
A few more things!
